tools/PI/DevHome.PI/BarWindow.xaml

```diff
@@ -126,19 +113,38 @@
                                     </MenuFlyoutItem.Icon>
                                 </MenuFlyoutItem>
                             </MenuFlyout>
-                        </ItemsRepeater.ContextFlyout>
+                        </StackPanel.ContextFlyout>
+
+                        <Button
+                             x:Name="ManageExternalToolsButton"
+                             x:Uid="ManageExternalToolsButton"
+                             HorizontalAlignment="Center"
+                            Click="ManageExternalToolsButton_Click">
+                            <Button.ContextFlyout>
+                                <MenuFlyout x:Name="ExternalToolsMenu">
+                                    <!-- We can't databind to MenuFlyout, so we'll add a MenuFlyoutItem
+                                    for each external tool in codebehind here. -->
+                                </MenuFlyout>
+                            </Button.ContextFlyout>
+                            <TextBlock Text="&#xEC7A;"/>
+                        </Button>
 
-                        <DataTemplate x:DataType="helpers:ExternalTool">
-                            <Button
-                                Click="ExternalToolButton_Click"
-                                HorizontalAlignment="Center" Width="45" Height="45"
-                                ToolTipService.ToolTip="{x:Bind Name, Mode=OneWay}"
-                                Tag="{x:Bind}"
-                                PointerPressed="ExternalToolButton_PointerPressed">
-                                <Image Source="{x:Bind ToolIcon, Mode=OneWay}"/>
-                            </Button>
-                        </DataTemplate>
-                    </ItemsRepeater>
+                        <ItemsRepeater
+                            x:Name="ExternalToolsRepeater"
+                            ItemsSource="{x:Bind helpers:ExternalToolsHelper.Instance.ExternalTools, Mode=OneWay}"
+                            Layout="{StaticResource ExternalToolsHorizontalLayout}">
+                            <DataTemplate x:DataType="helpers:ExternalTool">
+                                <Button
+                                    Click="ExternalToolButton_Click"
+                                    HorizontalAlignment="Center" Width="45" Height="45"

```

Comment:
This is going to be an accessibility bug. Please do not set hard-coded Height and Width on buttons. At the very least, use MinWidth. I'm a little concerned about setting Height at all, since this button will now look just a little off from the rest of Dev Home buttons (there's no screenshot with this PR, so if this isn't the case, I can't tell).

Reply:
Sorry, I didn't realize this was an image. It's such a common mistake with text that I jumped the gun.

Reply:
OK, thanks. I'll move the width/height to the Image itself, should make it clearer.